packages/block-editor/src/components/line-height-control/index.js

```diff
@@ -2,80 +2,92 @@
  * WordPress dependencies
  */
 import { __ } from '@wordpress/i18n';
-import { TextControl } from '@wordpress/components';
+import {
+	Button,
+	TextControl,
+	CustomSelectControl,
+} from '@wordpress/components';
 import { ZERO } from '@wordpress/keycodes';
 
 /**
  * Internal dependencies
  */
-import {
-	BASE_DEFAULT_VALUE,
-	RESET_VALUE,
-	STEP,
-	isLineHeightDefined,
-} from './utils';
+import { RESET_VALUE, STEP } from './utils';
 
-export default function LineHeightControl( { value: lineHeight, onChange } ) {
-	const isDefined = isLineHeightDefined( lineHeight );
+const DEFAULT_OPTION = {
+	key: 'default',
+	name: __( 'Default' ),
+	value: undefined,
+};
+const CUSTOM_OPTION = {
+	key: 'custom',
+	name: __( 'Custom' ),
+	value: 'custom',
+};
 
+export default function LineHeightControl( { presetValues, value, onChange } ) {
 	const handleOnKeyDown = ( event ) => {
 		const { keyCode } = event;
 
-		if ( keyCode === ZERO && ! isDefined ) {
+		if ( keyCode === ZERO ) {
```

Comment:
What this code does is that when the user types a 0, the whole line-height value is reset to 0. This is a nice little detail! However, I also wonder: this is preventing the user from introducing any value that includes a 0, as it'll be reset. Examples of values a user is not able to introduce: "10", "1.05", etc. Unless there's something that I'm missing I presume this restriction should be removed.

Reply:
@ItsJonQ what do you think of this? can we remove this logic?

Reply:
I believe so!! We'll be able to remove the manual keyDown handling once we migrate to using `<NumberControl />`

Reply:
So I've removed the existing `handleOnKeyDown` logic in https://github.com/WordPress/gutenberg/pull/23177/commits/98b1468e3695ea9c5aeb40ec5998ae9956c50842 and this seems to work fine for me as it is. At least I don't know how to break it! This is what I tried: click the arrow up/down, keyboard interactions (both starting with an existing value and with no value).